### ISSUE TITLE
Target stuck

### ISSUE THREAD

**Issue by MohitSethi99** [EpicGamesExt/raddebugger#186]
Not sure how I should explain this but here we go.

I'm using it with my work project and most of the time it fails to launch the target, The target application seems to be stuck during the initialisation phase with some modules loaded.

One thing that caught my attention was the dbghelp.dll that is loaded has raddbgi file of size 0 bytes (yes we are using some version of dbghelp.dll with a pdb)

Also, this issue is not new and has been present since the first release.

Another note: It works fine with vs debugger and remedybg

**Comment by ryanfleury** [comment]
That RADDBGI file might be relevant. Is it possible to send me the DLL/PDB for the dbghelp.dll?

It also might not be relevant, though, although given that it has been happening since the beginning, it is probably not due to my recent process control passes. The next big thing I'm hoping to get done is logging/diagnosis systems, so pretty soon you should be able to send me a log showing callstacks and logs, so that I can see e.g. what the control thread is doing.

**Comment by MohitSethi99** [comment]
That sounds great! About files, can you contact me on any of the platforms listed on my github profile?

**Comment by mmozeiko** [comment]
I'm not sure if that is supposed to be fixed, but raddbg --convert did not work for me on stripped any pdb's since forever - the ones you get from MS symbol servers. It was always crashing for me, thus producing 0 bytes converted output. Initially it showed call stack, but now due to --quiet it just silently fails.

**Comment by mmozeiko** [comment]
Here is kernel32.exe & pdb you can download:
https://msdl.microsoft.com/download/symbols/kernel32.dll/9EC9DA27BD000/kernel32.dll
https://msdl.microsoft.com/download/symbols/kernel32.pdb/B07C97792B439ABC0DF83499536C7AE51/kernel32.pdb

Running `raddbg.exe --convert --pdb:kernel32.pdb --exe:kernel32.dll --out:kernel32.rdi` produces:

```
Call stack:
1. [0x7ff6671831cd] p2r_convert +8669, raddbgi_from_pdb.c line 2854
2. [0x7ff667149082] entry_point +6546, raddbg_main.cpp line 382
3. [0x7ff667143512] main_thread_base_entry_point +562, base_entry_point.c line 81
4. [0x7ff6672dcda0] w32_entry_point_caller +512, os_core_win32.c line 1790
5. [0x7ff6671336b6] wWinMain +54, os_core_win32.c line 1802
6. [0x7ff6672f17ce] __scrt_common_main_seh +262, exe_common.inl line 288
7. [0x7ffdd2b17344] BaseThreadInitThunk +20
8. [0x7ffdd3be26b1] ntdll

Version: 0.9.9 [1c3792a]
```

**Comment by MohitSethi99** [comment]
Yeah, it makes sense cos it was producing 0 bytes output for all the dlls with fast debug, turning it off works but this one file is always 0 bytes and I don't think I can do anything about it since it's part of windows sdk.

**Comment by MohitSethi99** [comment]
I also tried deleting the PDB and RADDBGI files, just keeping the DLL but no luck.

**Comment by MohitSethi99** [comment]
So I started debugging the debugger debugging same target and found that control thread is stuck in this loop and never finds the good_read
![image](https://github.com/EpicGamesExt/raddebugger/assets/29519295/45766577-d28e-48ad-a049-65070ac26251)

**Comment by MohitSethi99** [comment]
Also, did some digging and yes for the time when it gets stuck, the target is setting up threads/fibers

**Comment by MohitSethi99** [comment]
Can confirm, removed the check, and target process is launching, also thread names in scheduler window are fine as well.
![image](https://github.com/EpicGamesExt/raddebugger/assets/29519295/300bef99-c111-4a0a-8656-addc6f0d4696)

**Comment by MohitSethi99** [comment]
I believe it should be:
```c
if(good_read)
{
    ...
}
else
{
    read_addr += 256;
}
```

**Comment by ryanfleury** [comment]
Thanks for looking into this; fixed in d560f8c734ba87fd1ea79e2dd5fefb6e3031fe00.
